mteb/models/rerankers_custom.py

```diff
@@ -0,0 +1,251 @@
+
+        self.model.eval()
+
+    @torch.inference_mode()
+    def predict(self, input_to_rerank, **kwargs):
+        queries, passages, instructions = list(zip(*input_to_rerank))
+        if instructions is not None and instructions[0] is not None:
+            queries = [f"{q} {i}".strip() for i, q in zip(instructions, queries)]
+
+        tokens = self.tokenizer(
+            queries,
+            passages,
+            padding=True,
+            truncation="only_second",
+            return_tensors="pt",
+            max_length=self.max_length,
+        ).to(self.device)
+        output = self.model(**tokens)[0]
+        batch_scores = torch.nn.functional.log_softmax(output, dim=1)
+        return batch_scores[:, 1].exp().tolist()
+
+
+class JinaReranker(RerankerWrapper):
+    name = "Jina"
+
+    def __init__(
+        self,
+        model_name_or_path="jinaai/jina-reranker-v2-base-multilingual",
+        torch_compile=False,
+        **kwargs,
+    ):
+        super().__init__(model_name_or_path, **kwargs)
+        if not self.device:
+            self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        model_args = {}
+        if self.fp_options:
+            model_args["torch_dtype"] = self.fp_options
+
+        self.model = CrossEncoder(
+            model_name_or_path,
+            automodel_args={"torch_dtype": "auto"},
+            trust_remote_code=True,
+        )
+
+    def predict(self, input_to_rerank, **kwargs):
+        queries, passages, instructions = list(zip(*input_to_rerank))
+        if instructions is not None and instructions[0] is not None:
+            queries = [f"{q} {i}".strip() for i, q in zip(instructions, queries)]
+
+        if self.first_print:
+            logger.info(f"Using {queries[0]}")
+            self.first_print = False
+
+        sentence_pairs = list(zip(queries, passages))
+        scores = self.model.predict(sentence_pairs, convert_to_tensor=True).tolist()
+        return scores
+
+
+def _loader(wrapper: type[RerankerWrapper], **kwargs) -> Callable[..., Encoder]:
+    _kwargs = kwargs
+
+    def loader_inner(**kwargs: Any) -> Encoder:
+        return wrapper(**_kwargs, **kwargs)
+
+    return loader_inner()
+
+
+monobert_large = ModelMeta(
+    loader=partial(
+        _loader,
+        wrapper=MonoBERTReranker,
+        model_name_or_path="castorini/monobert-large-msmarco",
+        fp_options="bfloat16",
+    ),
+    name="castorini/monobert-large-msmarco",
+    languages=["eng_Latn"],
+    open_source=True,
+    revision="0a97706f3827389da43b83348d5d18c9d53876fa",
+    release_date="2020-05-28",
+)
+
+# languages unclear: https://huggingface.co/jinaai/jina-reranker-v2-base-multilingual/discussions/28
+jina_reranker_multilingual = ModelMeta(
+    loader=partial(
+        _loader,
+        wrapper=JinaReranker,
+        model_name_or_path="jinaai/jina-reranker-v2-base-multilingual",
+        fp_options="bfloat16",
```

Comment:
Isn't that an odd default?

Reply:
I realize now this may not be common, but my understanding was that bfloat16 was basically identical in scores to fp32 and doesn't have the reliability issues of fp16.  That may be incorrect though. Would you suggest a different fp option?

I was not aware of #1297, but I think that wouldn't apply for rerankers? Could be wrong

Reply:
LMK if I should change it @KennethEnevoldsen!

Reply:
I think this should be changed because T4 (in kaggle, colab) doesn't support bfloat16

Reply:
Ah that's helpful.  So fp32 then? Or fp16?

Reply:
I switched to FP16.  If we want FP32, lmk otherwise I think we're good

Reply:
FP16 is good